tools/osv-scanner/cmd/scanner.go

```diff
@@ -119,18 +155,78 @@ func printResults(query osv.BatchedQuery, resp *osv.BatchedResponse) {
 }
 
 // TODO(ochang): Machine readable output format.
-// TODO(ochang): Ability to specify type of input.
 func main() {
-	flag.Usage = usage
-	flag.Parse()
-
 	var query osv.BatchedQuery
 
-	for _, arg := range flag.Args() {
-		if err := scan(&query, arg); err != nil {
-			log.Printf("scan failed: %v\n", err)
-			return
-		}
+	app := &cli.App{
+		Name:  "osv-scanner",
+		Usage: "scans various mediums for dependencies and matches it against the OSV database",
+		Commands: []*cli.Command{
+			{
+				Name:  "docker",
```

Comment:
I'm assuming that these let us do:

```bash
$ osv-scanner docker ...
$ osv-scanner lockfile ...
```
etc

Maybe instead we build a more flexible and extensible interface like so:

```bash
$ osv-scanner scan  -docker docker/image -docker docker/image2 -lockfile blah -sbom blah.sbom  <dir/path>
````

That is, we can pass multiple types in a single command. Paths can be passed with either an explicit type (e.g. "-sbom"), or where a type **isn't** supplied, we try to do our automated directory walking/scanning.

Reply:
Ah good idea, I'll change it to this now